src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -182,12 +179,9 @@ where
 
     let mut tokens = vec![];
 
-    let (source_upper_tx, source_upper_rx) = config.metrics.get_instrumented_source_channel(
```

Comment:
is it a big deal to lose metrics on this channel?

Reply:
This was previously an unbounded channel and the metrics were there to detect it holding too many values. This PR replaces it with a tokio watch channels which holds onto at most one value

Reply:
Doesn't that somewhat change the semantics of the existing implementation, though? Now we will reclock against the latest recorded frontier of the source rather than going in order of the individual uppers received from the source over this channel. This seems fine but I am curious if that's the intent, too